xdsl/backend/riscv/register_allocation.py

```diff
@@ -194,12 +194,18 @@ def allocate_registers(self, module: ModuleOp) -> None:
         """
         for op in module.walk():
             if isinstance(op, riscv_scf.ForOp):
-                for arg in op.body.block.args:
+                for i, arg in enumerate(op.body.block.args):
```

Comment:
I forgot this change was in this branch... I think I'll separate it out into another PR